Release r1.1 (Fall'25 M3)

ToshiWakayama-KDDI opened the pull request:
#### What type of PR is this?

* subproject management

#### What this PR does / why we need it:

Publication of Fall'25 M3 release candidate v0.4.0-rc.1

#### Which issue(s) this PR fixes:

<!-- Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`. -->

Fixes #N/A

#### Special notes for reviewers:

* This is the first release after the repository split, so the release tag is r1.1.
* CHANGELOG.md currently does not have the past log of CHANGELOG.md in the old repository, but it is planned to add the past log to CHANGELOG.md by M4 to record the whole history. (If this is fine with the Release Management.)
* README.md needs to be modified, but it will be done by a separate PR as an initial task of the new repository.  (If this is fine with the Release Management.)

#### Changelog input

```
 release-note
- Publication of Fall'25 M3 release candidate v0.4.0-rc.1
```

#### Additional documentation 

None

GillesInnov35 reviewed (approved):
Thanks Toshi

hdamker commented:
> * README.md needs to be modified, but it will be done by a separate PR as an initial task of the new repository.  (If this is fine with the Release Management.)

Please update the "Release Information" section of the README.md here within the release PR (as this shouldn't happen neither before nor after the release). All other changes (scope, contribution section) you can and should done in separate PR (before the release).

ToshiWakayama-KDDI commented:
Hi @hdamker , cc @GillesInnov35 , @fernandopradocabrillo , All,

Thank you, Herbert.  I have updated the "Release Information" section of the README.md.  Also, I have updated the scope and the contribution sections of the README.md, as it seemed not a big deal. 

Thanks,
Toshi

ToshiWakayama-KDDI commented:
Hi @hdamker , @tanjadegroot , @GillesInnov35 , @fernandopradocabrillo ,

I have just added the history of changelog to CHANGELOG.md, by copy-and-pasting from the old repository.  It does not seem working to jump from the table-of-contents to the tags in the CHANGELOG.md.  Better to think of another way...

Best regards,
Toshi

ToshiWakayama-KDDI commented:
So, what else could we do?  For example, 

a. Just put the link to the old CHANGELOG.md in the new CHANGELOG.md file.
b. Remove the tag jump from the history part of the new CHANGELOG.md.

WDYT?

Best regards,
Toshi

hdamker commented:
> I have just added the history of changelog to CHANGELOG.md, by copy-and-pasting from the old repository. It does not seem working to jump from the table-of-contents to the tags in the CHANGELOG.md. Better to think of another way...

As I have stated in other discussions about the CHANGELOG.md I wouldn't have copied the history from the previous repository over. A clear note and link to the previous repository and CHANGELOG.md would have done the job and avoid such confusing situation like two time a r1.1 in the CHANGELOG.md.

BTW: we will need to restructure the CHANGELOG.md anyway at some point of time as they will get too long, e.g. into separate files for each release cycle like Kubernetes is doing it (e.g. https://github.com/kubernetes/kubernetes/tree/master/CHANGELOG).

cc: @tanjadegroot

ToshiWakayama-KDDI commented:
Hi @GillesInnov35 , @fernandopradocabrillo , all, 

Release Information part in the README.md is now under discussion.  Please see https://github.com/camaraproject/KnowYourCustomerMatch/pull/27#issuecomment-3034186152.

Thanks,
Toshi

FabrizioMoggio commented:
About the readme.md file. If you look at: https://github.com/camaraproject/QualityOnDemand/blob/7ae997582948f3b6a9e77870c5711acb7b75f24f/README.md

They have a sentence announcing the new pre-release:

NEW: Pre-release [r3.1](https://github.com/camaraproject/QualityOnDemand/releases/tag/r3.1) with quality-on-demand v1.1.0-rc.2, qos-profiles v1.1.0-rc.2 and qos-provisioning v0.3.0-rc.1 is available.

And a section linking the YAML for the last **Public** Release:

In our situation, because we have a new Repo, we just need the sentence for the NEW pre-release and refer to the old repo for the stable release. I did like this here: https://github.com/camaraproject/KnowYourCustomerMatch/blob/ce3585b79e009cb8bc281b46f6063d99968ada93/README.md

@ToshiWakayama-KDDI , @fernandopradocabrillo, @hdamker , @tanjadegroot What do you think?

ToshiWakayama-KDDI commented:
Thanks, @FabrizioMoggio .  Fine with me, as I commented https://github.com/camaraproject/KnowYourCustomerMatch/pull/27#issuecomment-3043659408, and I have modified the README.md of the kyc-fill-in proposal.  
->And, @GillesInnov35 , please take care of kyc-age-verification, if possible. 

And also I have modified the test .feature file for the x-correlator related sentence in the background sectionl, as I commented https://github.com/camaraproject/KnowYourCustomerMatch/pull/27#issuecomment-3043690855.  Please, could you @GillesInnov35 take care of kyc-age-verification, and could you @FabrizioMoggio take care of kyc-match?

Many thanks,
Toshi

ToshiWakayama-KDDI commented:
> hi @ToshiWakayama-KDDI , yes I take the action. Thanks there is one **_"_** missing at the end of the sentence ` And the header "x-correlator" complies with the schema at "#/components/schemas/XCorrelator`

Thanks @GillesInnov35 .  I have just corrected that.

Best regards,
Toshi

ToshiWakayama-KDDI commented on README.md:
```diff
@@ -16,31 +16,27 @@ Incubating API Repository to evolve and maintain the definitions and documentati
-* The API(s) provide(s) the API consumer with the ability to:  
-  * §Short API scope description§ !! Add here a scope description according to the API proposal
+* The API provides the API consumer with the ability to:  
+  * request and receive the information for a particular end user, which on file (and verified) by the end user's Operator in their own KYC records, in order for the service provider (API consumer) to confirm the accuracy of the information and provide a specific service to the end user.
```
Thank you, @jlurien .

ToshiWakayama-KDDI commented on CHANGELOG.md:
```diff
@@ -0,0 +1,51 @@
+  - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/KnowYourCustomerFill-in/r1.1/code/API_definitions/kyc-fill-in.yaml&nocors)
+  - [View it on Swagger Editor](https://editor.swagger.io/swagger-ui/?url=https://raw.githubusercontent.com/camaraproject/KnowYourCustomerFill-in/r1.1/code/API_definitions/kyc-fill-in.yaml&nocors)
+
+**Changed**
```
Thank you, @jlurien .  I have just updated CHANGELOG.md of this PR.  Hope you could check it.   Thanks.

ToshiWakayama-KDDI commented:
Thanks, @jlurien .

> 
> kyc-fill-in: ExternalDocs - Missing externalDocs object 

Regarding this, I think the URL Link for Readiness Checklist No.13 API description (for Marketing) should be corrected. 
Currently, for Fill-in, the link is 'https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/137592951/KnowYourCustomerFill-in+API+description'.
It should be changed to the one like kyc-match's 'https://lf-camaraproject.atlassian.net/wiki/x/RwAzC7'. 

Could you advise me how I can check the link for Fill-in?  I do not know the bold part for Fill-in: https://lf-camaraproject.atlassian.net/wiki/ **x/RwAzC7** .

Many thanks,
Toshi

ToshiWakayama-KDDI commented:
Hi @jlurien ,
> 
> kyc-fill-in Tests: Test Operation IDs - Test references unknown operation KYC_Fill-In

Regarding this, I think L.29 in the test .feature file is currently `When the HTTP "POST" request "KYC_Fill-In" is sent`, and Fill-In's '**In**' should be corrected to '**in**'.  Am I correct?

Many thanks,
Toshi

ToshiWakayama-KDDI commented:
Hi @jlurien ,

I saw your comment to KYC Match PR #27 ( https://github.com/camaraproject/KnowYourCustomerMatch/pull/27#pullrequestreview-2997128631 )

> Links in the spec must refer to this release, for example: src="[https://raw.githubusercontent.com/camaraproject/KnowYourCustomer/main/documentation/API_documentation/assets/kyc-match_flow.png">](https://raw.githubusercontent.com/camaraproject/KnowYourCustomer/main/documentation/API_documentation/assets/kyc-match_flow.png%22%3E) points to main

I think this comment should apply to KYC Fill-in as well.  If you agree, I will correct the link for Fill-in, as same as Fabrizio did for Match.  WDYT?

Many thanks,
Toshi

jlurien commented on CHANGELOG.md:
```diff
@@ -0,0 +1,60 @@
+* kyc-fill-in v0.4.0-rc.1
+
+The API definition(s) are based on
+* Commonalities v0.6.0
```
```suggestion
* Commonalities v0.6.0-rc.1 (r3.2)
```

jlurien commented on CHANGELOG.md:
```diff
@@ -0,0 +1,60 @@
+
+The API definition(s) are based on
+* Commonalities v0.6.0
+* Identity and Consent Management v0.4.0
```
```suggestion
* Identity and Consent Management v0.4.0-rc.1 (r3.2)
```

jlurien commented on CHANGELOG.md:
```diff
@@ -0,0 +1,60 @@
+
+## kyc-fill-in v0.4.0-rc.1
+
+**kyc-fill-in v0.4.0-rc.1 is the release-candidate version for v0.4.0 of the CAMARA KnowYourCustomer Fill-in API**
```
```suggestion
**kyc-fill-in v0.4.0-rc.1 is the first release-candidate version for v0.4.0 of the CAMARA KnowYourCustomer Fill-in API**
```

jlurien commented on documentation/API_documentation/kyc-fill-in-API-Readiness-Checklist.md:
```diff
@@ -1,22 +1,22 @@
 | 11 | Change log updated                           |   M   |         M         |    M    |    M   |  Y   | [link](/CHANGELOG.md) |
 | 12 | Previous public release was certified        |   O   |         O         |    O    |    M   |  N   |      |
-| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |      | [wiki link](https://lf-camaraproject.atlassian.net/wiki/xxx) |
+| 13 | API description (for marketing)              |   O   |         O         |    M    |    M   |  Y   | [wiki link](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/137592951/KnowYourCustomerFill-in+API+description) |
```
you can add just the short generated link to https://lf-camaraproject.atlassian.net/wiki/x/d4AzC, which is more independent of changes in page naming

ToshiWakayama-KDDI replied:
Thank you so much, @jlurien .  I have just replaced the link with what you have given. 

By the way, how could you get the short link?

Best regards,
Toshi

hdamker replied:
Two remarks here:
* Since our change to the Atlassion cloud are also the "long" links robust against page naming changes. The relevant part is in the example https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/137592951/ ... the rest can change.
* The "short" or "share" links can be generated with the combined Share/Link button in upper right corner of each page - just click on the right part of it:

<img width="166" height="59" alt="Screenshot 2025-07-16 at 17 47 07" src="https://github.com/user-attachments/assets/bac81d5a-a839-4354-b79a-e2e3f6ab18ab" />

cc: @wrathwolf do we need still to use the "share" links or are the others also ok?

ToshiWakayama-KDDI replied:
Thank you, @hdamker !  It's like magic.

jlurien reviewed (commented):
Some additional comments and suggestions. Almost ready

jlurien reviewed (approved):
Approved on behalf of @camaraproject/release-management_reviewers

ToshiWakayama-KDDI commented:
Hi @GillesInnov35 , @fernandopradocabrillo , 

Now this PR #19 is ready to be merged, but it seems approval from either of you is still needed.  Could you approve this PR, please?

Many thanks,
Toshi

GillesInnov35 reviewed (approved):
LGTM. Thanks Toshi

hdamker commented:
@ToshiWakayama-KDDI please create the GitHub release directly after merging the release PR (so that the tag is on the merge commit). If you have questions or need support with that, just let me know.
